homeassistant/components/dsmr/sensor.py

```diff
@@ -750,25 +732,25 @@ def __init__(
             self._attr_unique_id = f"{device_serial}_{entity_description.key}"
 
     @callback
-    def update_data(self, telegram: dict[str, DSMRObject] | None) -> None:
+    def update_data(self, telegram: Telegram | None) -> None:
         """Update data."""
         self.telegram = telegram
         if self.hass and (
-            telegram is None or self.entity_description.obis_reference in telegram
+            telegram is None
+            or hasattr(telegram, self.entity_description.obis_reference)
```

Comment:
@frenck : If I change the obis_reference to a lambda function, it's not possible to have this hasattr check.
How do you see it? Still use getattr? Or next to obis_reference add some 'obis_reference_str' with string value of the obis_reference just to be able to check if the object has this attribute?